File: 05/exercise1-objc/car.m
#import "car.h"

@interface Car()

@end

@implementation Car

-(id) Name: (NSString *) name MaxSpeed: (int) maxSpeed Color: (NSString *) color AmountOfEngineValve: (int) engine {
	self = [super init];
	self.name = name;
	self.color = color;
	self.amountOfEngineValve = engine;
	self.maxSpeed = maxSpeed;
	return self;
}

-(void) printDetail {
	printf("%s is %s and has a %d-valve engine. MAX SPEED = %d mph.\n", [self.name UTF8String], [self.color UTF8String], self.amountOfEngineValve, self.maxSpeed);
}

@end

File: 05/exercise1/car.cpp
#include "car.h"

using namespace std;

Car::Car() {
	this->name_ = "default";
	this->color_ = "black";
	this->maxSpeed_ = 90;
	this->amountOfEngineValve_ = 4;
}

Car::Car(string name, int maxSpeed, string color, int amountOfEngineValve) {
	this->name_ = name;
	this->color_ = color;
	this->maxSpeed_ = maxSpeed;
	this->amountOfEngineValve_ = amountOfEngineValve;
}

Car::~Car() {
	
}

void Car::printDetail() {
	cout << this->name_ << " is " << this->color_ << " and has a " << this->amountOfEngineValve_ << "-valve engine. MAX SPEED = " << this->maxSpeed_ << " mph." << endl;
}